janusgraph-solr/src/main/java/org/janusgraph/diskstorage/solr/SolrIndex.java

```diff
@@ -386,6 +386,13 @@ public void mutate(Map<String, Map<String, IndexMutation>> mutations, KeyInforma
                             doc.setField(v, isNewDoc ? adds.get(v) :
                                 new HashMap<String, Object>(1) {{put(solrOp, adds.get(v));}}
                             );
+                            // For textstring mapping, we create two indexes one for text and one for string
+                            if (hasDualStringMapping(keyInformation)) {
+                                String fieldName = getDualMappingName(v);
```

Comment:
elide the fieldName local variable and instead do getDualMappingName(v) directly on line 392